service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalogHandler.java

```diff
@@ -587,7 +604,17 @@ public Optional<LoadTableResponse> loadTableIfStale(
     }
 
     LoadTableResponse rawResponse = CatalogHandlers.loadTable(baseCatalog, tableIdentifier);
-    return Optional.of(filterResponseToSnapshots(rawResponse, snapshots));
+    Optional<LoadTableResponse> optionalLoadTableResponse = Optional.of(filterResponseToSnapshots(rawResponse, snapshots));
+    if (tableEntity != null
```

Comment:
This needs to be feature flagged off. Also, this only handls conversion to Iceberg. To handle conversion to other formats, you'll need to look at `GenericTableCatalogHandler`.

Reply:
I added changes in the `GenericTableCatalogHandler`, trying to understand though why this needs to be flagged off?